ksp_plugin/plugin.cpp

```diff
@@ -478,6 +514,34 @@ bool Plugin::is_dirty(not_null<Vessel*> const vessel) const {
   return dirty_vessels_.count(vessel) > 0;
 }
 
+bool Plugin::has_predicted_vessel() const {
+  return predicted_vessel_ != nullptr;
+}
+
+bool Plugin::has_predictions() const {
+  if (system_predictions_.empty()) {
+    CHECK(prediction_ == nullptr);
+    return false;
+  } else {
+    CHECK_NOTNULL(prediction_);
+    CHECK(has_predicted_vessel());
+    return true;
+  }
+}
+
+void Plugin::clear_predictions() {
+  if (has_predictions()) {
+    predicted_vessel_->mutable_prolongation()->DeleteFork(&prediction_);
+    for (auto it = system_predictions_.begin();
+         it != system_predictions_.end();
+         it = system_predictions_.erase(it)) {
```

Comment:
C++11 makes `it = v.erase(it)` equivalent to `v.erase(it++)`.
The latter is more concise, the former seems to be intended by the standard, I'm not sure which one we prefer.